test/infrastructure/docker/internal/docker/machine.go

```diff
@@ -416,28 +416,22 @@ func (m *Machine) SetNodeProviderID(ctx context.Context) error {
 }
 
 func (m *Machine) getKubectlNode(ctx context.Context) (*types.Node, error) {
-	// collect info about the existing controlplane nodes
+	// collect info about the existing nodes
 	filters := container.FilterBuilder{}
 	filters.AddKeyNameValue(filterLabel, clusterLabelKey, m.cluster)
-	filters.AddKeyNameValue(filterLabel, nodeRoleLabelKey, constants.ControlPlaneNodeRoleValue)
 
 	kubectlNodes, err := listContainers(ctx, filters)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
-	if len(kubectlNodes) == 0 {
-		return nil, nil
-	}
-	// Return the first node that is not the current machine.
-	// The assumption being made is that the existing control planes will already be ready.
-	// This is true when we are using kubeadm control plane.
+	// Return the node matching the current machine, required to patch itself using its kubelet config
 	for _, node := range kubectlNodes {
-		if node.Name != m.container.Name {
+		if node.Name == m.container.Name {
 			return node, nil
 		}
 	}
-	// This will happen when the current machine is the only machine.
-	return kubectlNodes[0], nil
+
+	return nil, nil
```

Comment:
We should avoid returning `nil, nil`, shouldn't we have an error if we haven't found the node?

Reply:
Good point. We also did this before, but let's fix / improve it

Reply:
Address with d4b2a03a9, let me know if it should be squashed in a single commit.

Reply:
Looks good. Yup, please squash

Reply:
Thanks, squashed.